src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -266,5 +266,24 @@ public void GetFullPath_CoreTests(string path, string expected)
         {
             Assert.Equal(expected, Path.GetFullPath(path));
         }
+
+        [Theory,
+        InlineData(@"C:\folder\", @"C:\folder"),
+        InlineData(@"C:/folder/", @"C:/folder"),
+        InlineData(@"/folder/", @"/folder"),
+        InlineData(@"\folder\", @"\folder"),
+        InlineData(@"/folder//", @"/folder/"),
+        InlineData(@"\folder\\", @"\folder\"),
+        InlineData(@"C:\", @"C:\"),
+        InlineData(@"C:/", @"C:/"),
+        InlineData(@"", @""),        
+        InlineData(@"/", @"/"),
+        InlineData(@"\", @"\"),
+        InlineData(null, null)]
+        public void TrimEndingDirectorySeparator_CoreTests(string path, string expected)
+        {
+            Assert.Equal(expected, Path.TrimEndingDirectorySeparator(path));
```

Comment:
I'd suggest changing this to:
```C#
string trimmed = Path.TrimEndingDirectorySeparator(path);
Assert.Equal(expected, trimmed);
Assert.Same(trimmed, Path.TrimEndingDirectorySeparator(trimmed));
```
to validate that operating on something that's already been trimmed returns the same object.